### DISCUSSION TITLE
Overlay placement

### DISCUSSION THREAD

**Discussion by tobi-or-not-tobi** [floating-ui/floating-ui#2767] [Ideas]
Cool library, thanks for building. I'm new to it, so perhaps this exists. I'm looking for a placement option that will place an overlay over an existing element. Is this already possible? If not, are there low level floating-UI APIs available that I can use to build it myself?

I could imagine a new placement "overlay", but it would fail on a few things. E.g. the arrow is n/a, the 2nd argument in `computePosition()` is obsolete, etc.

**Comment by atomiks** [comment]
First example here is "overlay": https://floating-ui.com/docs/offset#creating-custom-placements
If the size needs to match too, `size()` middleware works

**Reply by tobi-or-not-tobi** [reply]
That's very helpful @atomiks, thanks!

I started using offset, and manipulated the width/height - feels a bit clunky. I then tried to use `size()`. You can find the trail of both approaches in this snippet:

```ts
// let w: number;
// let h: number;

computePosition(this, this.highlight, {
  placement: 'bottom-start',
  middleware: [
    offset(({ rects }) => {
      // w = rects.reference.width;
      // h = rects.reference.height;
      return -rects.reference.height;
    }),
    size(({ elements, rects }) => {
      return Object.assign(elements.floating.style, {
        width: `${rects.reference.width}px`,
        height: `${rects.reference.height}px`,
      });
    }),
  ],
}).then(({ x, y }: { x: number; y: number }) => {
  this.style.setProperty('--x', `${x}px`);
  this.style.setProperty('--y', `${y}px`);
  // this.style.setProperty('--h', `${h}px`);
  // this.style.setProperty('--w', `${w}px`);
});
```

I'd appreciate if you see a more elegant solution.

**Reply by atomiks** [reply]
Since you don't need `flip` and other middleware, you can just set the size beforehand:

```js
// Make the highlight the same size as the reference:
const rect = this.getBoundingClientRect();
this.highlight.style.setProperty('--h', `${rect.height}px`);
this.highlight.style.setProperty('--w', `${rect.width}px`);

computePosition(this, this.highlight, {
  middleware: [
    offset(({ rects }) => -rects.floating.height),
  ],
}).then(({ x, y }: { x: number; y: number }) => {
  this.style.setProperty('--x', `${x}px`);
  this.style.setProperty('--y', `${y}px`);
});
```

**Reply by tobi-or-not-tobi** [reply]
Cool, thx. I was expecting more magic that could influence the box model such as `box-sizing`, let's see if this is enough. 
Thanks for you help!
